src/Exception/AbstractTransformerException.php

```diff
@@ -0,0 +1,61 @@
+<?php
+declare(strict_types=1);
+
+namespace Diaclone\Exception;
+
+use Diaclone\Transformer\AbstractTransformer;
+
+abstract class AbstractTransformerException extends TransformException
+{
+    private $invalidFields;
+    /** @var AbstractTransformer */
+    protected $transformer;
+
+    public function __construct($fields, AbstractTransformer $transformer = null, string $message = null)
```

Comment:
Type hint `array $fields` because getter expects it, or cast to array on getter.

Reply:
I had it like that because https://github.com/finicprint/diaclone/pull/34/files#diff-6513d4fb3b838d027c4dd410a08d3058L47 is checking for a string. However, I didn't see anywhere it was being used that way.

I would prefer to cast it as an array and get rid of that like that checks for the string, unless you know where it might be used as a string

Reply:
I'm ok with casting it as an array

Reply:
Ok, I'll update that and remove those setters